hoomd/md/constrain.py

```diff
@@ -336,237 +189,103 @@ class rigid(ConstraintForce):
-            for o in orientations_list:
-                o = tuple(o)
-                if len(o) != 4:
-                    hoomd.context.current.device.cpp_msg.error(
-                        "Particle orientation is not a 4-tuple.\n")
-                    raise RuntimeError(
-                        "Error setting up parameters for constrain.rigid()")
-                orientation_vec.append(
-                    _hoomd.make_scalar4(o[0], o[1], o[2], o[3]))
-        else:
-            for p in positions:
-                orientation_vec.append(_hoomd.make_scalar4(1, 0, 0, 0))
-
-        charge_vec = _hoomd.std_vector_scalar()
-        if charges is not None:
-            charges_list = list(charges)
-            for c in charges_list:
-                charge_vec.append(float(c))
-        else:
-            for p in positions:
-                charge_vec.append(0.0)
-
-        diameter_vec = _hoomd.std_vector_scalar()
-        if diameters is not None:
-            diameters_list = list(diameters)
-            for d in diameters_list:
-                diameter_vec.append(float(d))
-        else:
-            for p in positions:
-                diameter_vec.append(1.0)
-
-        # set parameters in C++ force
-        self.cpp_force.setParam(
-            type_id,
-            type_vec,
-            pos_vec,
-            orientation_vec,
-            charge_vec,
-            diameter_vec,
-        )
-
-    def create_bodies(self, create=True):
-        R"""Create copies of rigid bodies.
-
-        Args:
-            create (bool): When True, create rigid bodies, otherwise validate
-                existing ones.
-        """
-        self.cpp_force.validateRigidBodies(create)
-
-    def validate_bodies(self):
-        R"""Validate that bodies are well defined and prepare for the
-        simulation run.
-        """
-        self.cpp_force.validateRigidBodies(False)
-
-    ## \internal
-    # \brief updates force coefficients
-    def update_coeffs(self):
-        # validate copies of rigid bodies
-        self.create_bodies(False)
-
-
-class oneD(ConstraintForce):
-    R"""Constrain particles to move along a specific direction only
+    Set constituent particle types and coordinates for a rigid body.
 
     Args:
-        group (``hoomd.group``): Group on which to apply the constraint.
-        constraint_vector (list): [x,y,z] list indicating the direction that
-        the particles are restricted to
+        type_name (str): The type of the central particle
+        types (list): List of types of constituent particles
+        positions (list): List of relative positions of constituent
+            particles
+        orientations (list): List of orientations of constituent particles
+            (**optional**)
+        charge (list): List of charges of constituent particles
+            (**optional**)
+        diameters (list): List of diameters of constituent particles
+            (**optional**)
```

Comment:
```suggestion
        types (list[str]): List of types of constituent particles
        positions (list[ tuple[float, float, float] ]): List of relative positions of constituent
            particles
        orientations (list[ tuple[float, float, float, float]]): List of orientations of constituent particles
            (**optional**)
        charge (list[float]): List of charges of constituent particles
            (**optional**)
        diameters (list[float]): List of diameters of constituent particles
            (**optional**)
```
Fix the typing syntax. My suggestion might require some tweaks to render properly in sphinx and may not be wrapped properly.